pkg/pgmodel/migrations/sql/idempotent/base.sql

```diff
@@ -1054,94 +1071,138 @@ BEGIN
+                    WHERE delete_epoch <= deletion_epoch
+                        AND NOT EXISTS (
+                            SELECT 1 FROM SCHEMA_DATA.%1$I
+                            WHERE id = label_id LIMIT 1
+                        )
+            ), deleted_series AS (
+                DELETE FROM SCHEMA_DATA_SERIES.%1$I
+                WHERE delete_epoch <= deletion_epoch
+                    AND id IN (SELECT id FROM dead_series) -- concurrency means we need this qual in both
+                RETURNING id, labels
+            ), resurrected_series AS (
+                UPDATE SCHEMA_DATA_SERIES.%1$I
+                SET delete_epoch TO NULL
+                WHERE delete_epoch <= deletion_epoch
+                    AND id NOT IN (SELECT id FROM dead_series) -- concurrency means we need this qual in both
             )
-        ), deleted_series AS (
-          DELETE from SCHEMA_DATA_SERIES.%1$I
-          WHERE id IN (SELECT series_id FROM confirmed_drop_series)
-          RETURNING id, labels
-        )
-        SELECT ARRAY(SELECT DISTINCT unnest(labels) as label_id
-        FROM deleted_series)
-    $query$, metric_table, older_than, check_time) INTO label_array;
-
-    --needs to be a separate query and not a CTE since this needs to "see"
-    --the series rows deleted above as deleted.
-    --Note: we never delete metric name keys since there are check constraints that
-    --rely on those ids not changing.
-    EXECUTE format($query$
-    WITH confirmed_drop_labels AS (
-            SELECT label_id
-            FROM unnest($1) as labels(label_id)
-            WHERE NOT EXISTS (
-                 SELECT 1
-                 FROM  SCHEMA_DATA_SERIES.%1$I series_exists
-                 WHERE series_exists.labels && ARRAY[labels.label_id]
-                 LIMIT 1
+            SELECT ARRAY(SELECT DISTINCT unnest(labels) as label_id
+            FROM deleted_series
+        $query$, metric_table) INTO label_array;
+
+        --needs to be a separate query and not a CTE since this needs to "see"
+        --the series rows deleted above as deleted.
+        --Note: we never delete metric name keys since there are check constraints that
+        --rely on those ids not changing.
+        EXECUTE format($query$
+        WITH confirmed_drop_labels AS (
+                SELECT label_id
+                FROM unnest($1) as labels(label_id)
+                WHERE NOT EXISTS (
+                    SELECT 1
+                    FROM  SCHEMA_DATA_SERIES.%1$I series_exists
+                    WHERE series_exists.labels && ARRAY[labels.label_id]
+                    LIMIT 1
+                )
             )
-        )
-        DELETE FROM SCHEMA_CATALOG.label
-        WHERE id IN (SELECT * FROM confirmed_drop_labels) AND key != '__name__';
-    $query$, metric_table) USING label_array;
+            DELETE FROM SCHEMA_CATALOG.label
+            WHERE id IN (SELECT * FROM confirmed_drop_labels) AND key != '__name__';
```

Comment:
Are we not using the same epoching stuff here? Seems like we'd run into the same issues?

Reply:
Aren't labels not particular to a given series? I thought they were shared? But maybe I'm wrong...

Reply:
We never store label_ids in promscalem, so they're safe to delete as soon as there's no series that references them

Reply:
Really? no caching for that? Okay. and also, what I was really asking here was whether looking for usage of a label in a given series is going to work or if we need to look everywhere, which could be a bug in the old code...

Reply:
Ah. I see why now on not storing them in promscale memory, but I don't get the other bit...

Reply:
label has `UNIQUE(key, value)` which certainly __suggests__ they can be shared across series...
@cevian?